keras/engine/training.py

```diff
@@ -1124,7 +1124,11 @@ class indices (integers) to
             elif len(validation_data) == 3:
                 val_x, val_y, val_sample_weight = validation_data
             else:
-                raise
+                raise ValueError("When passing validation_data, "
+                                 "it must contain 2 (x,y) or 3 (x,y,weights) "
+                                 "items, however it contains %d items" %
+                                 len(validation_data))
```

Comment:
PEP8: quote character should be `'` for consistency. Add spaces after `,`. Replace `weights` with `sample_weights` for explicitness.